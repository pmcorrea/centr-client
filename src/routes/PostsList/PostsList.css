
.PostsList ul {
  list-style-type: none;
  margin: 0;
  padding: 10px 0px;
  padding-bottom: 80px;
}

.PostsList__add-button-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  position: relative;
  width: 100vw;
  height: 50px;
}

.PostsList__add-button {
  display: flex;
  font-size: 40px;
  color: black;
}

.AddPost_button {
  
}