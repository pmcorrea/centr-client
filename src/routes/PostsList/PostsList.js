import React, {Component} from 'react'

// Components and services
import Post from '../../components/Post/Post'
import MainContext from '../../contexts/MainContext'
import {getPostsForFolder} from '../../services/posts-helpers'

// Style
import './PostsList.css'

export default class PostsList extends Component {
	static contextType = MainContext;

	componentDidMount() {
		this.context.getData()
	}

	render() {
		const folderId = this.props.match.params.folderId;
		let posts = this.context.posts;

		if (folderId) {
			posts = posts ? getPostsForFolder(posts, parseInt(folderId)) : {};
		}

		return posts ? (
			<section className="PostsList">
				<p className="section_headers">Notes</p>
				
				<ul className="PostList_ul">
					{posts.map(post => (
						<li key={post.id}>
							<Post
								id={post.id}
								name={post.post_name}
								visibility={post.visibility}
								modified={post.modified}
								history={this.props.history}
							/>
						</li>
					))}
				</ul>
			</section>
		) : ('');
	}
}

PostsList.defaultProps = {
	posts: []
};
