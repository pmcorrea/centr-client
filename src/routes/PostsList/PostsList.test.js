import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import PostsList from './PostsList'

describe(`PostList component`, () => {
  const props = {
    posts: [
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Dogs",
        "modified": "2019-01-03T00:00:00.000Z",
        "folderId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Corporis accusamus placeat.\n \rUnde."
      },
      {
        "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Cats",
        "modified": "2018-08-15T23:00:00.000Z",
        "folderId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Eos\n \rlaudantium."
      },
      {
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Pigs",
        "modified": "2018-03-01T00:00:00.000Z",
        "folderId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Occaecati dignissimos\nvoluptatum nihil."
      },
      {
        "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Birds",
        "modified": "2019-01-04T00:00:00.000Z",
        "folderId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Eum culpa odit."
      },
    ]
  }

  it('renders a .PostList by default', () => {
    const wrapper = shallow(<PostsList />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Post in ul for each posts in array', () => {
    const ul = shallow(<PostsList {...props} />)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
