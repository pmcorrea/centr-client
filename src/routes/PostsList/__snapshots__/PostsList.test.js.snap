// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsList component renders a .PostsList by default 1`] = `""`;

exports[`PostsList component renders a Post in ul for each posts in array 1`] = `null`;
